### DISCUSSION TITLE
Delphi improvement

### DISCUSSION THREAD

**Discussion by mathias-burbach** [PlummersSoftwareLLC/Primes#372] [Improvement suggestions]
I managed to get the current implementation to achieve 2.2% more passes by avoiding an Exit call in TPrimeSieve.ClearBit by changing 

procedure TPrimeSieve.ClearBit(Index: Integer);
begin
  if (Index mod 2) = 0 then
    Exit;
  FBitArray[Index div 2] := False;
end;

to

procedure TPrimeSieve.ClearBit(Index: Integer);
begin
  if (Index mod 2) = 1 then
    FBitArray[Index div 2] := False;
end;

The Exit call causes more assembler code.

I am unsure how to get this improvement into the repository.

Salut,
  Mathias

**Comment by rbergen** [comment]
> I am unsure how to get this improvement into the repository.

With #264 already aiming to improve the existing Delphi solution, you could add a comment to that PR that points this out.
